Easydict/Feature/ViewController/Window/WindowManager/EZWindowManager.m

```diff
@@ -912,31 +911,42 @@ - (void)closeFloatingWindow {
     
     // stop playing audio
     [self.floatingWindow.queryViewController stopPlayingQueryText];
-    
     self.floatingWindow.titleBar.pin = NO;
+    
+    EZBaseQueryWindow *closingFloatingWindow = self.floatingWindow;
+    
+    /// !!!: Close window may call window delegate method `windowDidResignKey:`
+    /// `windowDidResignKey:` will call `closeFloatingWindowIfNotPinnedOrMain`
     [self.floatingWindow close];
     
     if (![self currentShowingSettingsWindow]) {
         // recover last app.
         [self activeLastFrontmostApplication];
     }
     
-    if ([EZMainQueryWindow isAlive]) {
-        [self.mainWindow orderBack:nil];
+    /// !!!: Maybe floating window has closed after calling [self.floatingWindow close], so we need to check it.
+    if (self.floatingWindow == closingFloatingWindow) {
```

Comment:
In what cases they are different?

Reply:
When enable main window, the fixed window is displayed for the first time, and then clicking on the blank space triggers `setDismissFixedWindowBlock` to close the window, which triggers `closeFloatingWindow` twice, the first time the floating window is the fixed window, the second time is the main window.

The second time is an exception, because the `floatingWindow` is dynamically changing according to the `floatingWindowType`, and its value may change in the middle.

<img width="1672" alt="image" src="https://github.com/tisfeng/Easydict/assets/25194972/027855f9-4009-442b-82b6-504c501a5a45">

Reply:
Since closing a window causes the floatingWindowType to change, I've added a windowType parameter to specify that a window should be closed, avoiding the problem mentioned above.

```objc
- (void)windowDidResignKey:(NSNotification *)notification {
//    NSLog(@"windowDidResignKey: %@", self);
    
    // Close floating window when losing focus if it's not pinned or main window.
    [EZWindowManager.shared closeFloatingWindowIfNotPinned:self.windowType exceptWindowType:EZWindowTypeMain];
}
```